### PR TITLE
ZENKO-2895 test-objects-tags-and-metadata-actions

### PR DESCRIPTION
**Pull request by romainlancia**

### CODE REVIEW

File: src/react/actions/__tests__/s3object.test.jsx [hunk 1]
```diff
@@ -75,6 +82,21 @@ describe('s3object actions', () => {
             fn: actions.toggleAllObjects(true),
             expectedActions: [dispatchAction.TOGGLE_ALL_OBJECTS_ACTION(true)],
         },
+        {
+            it: 'should return RESET_OBJECT_METADATA action',
+            fn: actions.resetObjectMetadata(),
+            expectedActions: [dispatchAction.RESET_OBJECT_METADATA_ACTION()],
+        },
+        {
+            it: 'should return GET_OBJECT_METADATA_SUCCESS action -> test without prefix parameter',
```
Comment (nicolas2bert) [inline src/react/actions/__tests__/s3object.test.jsx:95]: Can we provide a non-empty tags array? (type `TagSet`)

File: src/react/actions/__tests__/s3object.test.jsx [hunk 2]
```diff
@@ -264,6 +286,134 @@ describe('s3object actions', () => {
                 dispatchAction.NETWORK_END_ACTION,
             ],
         },
+        {
+            it: 'getObjectMetadata: should return expected actions -> test without prefix parameter',
+            fn: actions.getObjectMetadata(BUCKET_NAME, '', OBJECT_KEY),
+            storeState: initState,
+            expectedActions: [
+                gettingObjectMetadataNetworkAction,
+                dispatchAction.GET_OBJECT_METADATA_SUCCESS_ACTION(BUCKET_NAME, '', OBJECT_KEY, HEAD_OBJECT_RESPONSE, []),
```
Comment (nicolas2bert) [inline src/react/actions/__tests__/s3object.test.jsx]: Can we provide a non-empty tags array? (type `TagSet`)

File: src/react/actions/__tests__/s3object.test.jsx [hunk 3]
```diff
@@ -264,6 +286,134 @@ describe('s3object actions', () => {
...
+                gettingObjectTagsNetworkAction,
+                gettingObjectMetadataNetworkAction,
+                dispatchAction.GET_OBJECT_METADATA_SUCCESS_ACTION(BUCKET_NAME, '', OBJECT_KEY, HEAD_OBJECT_RESPONSE, []),
+                dispatchAction.NETWORK_END_ACTION,
+                dispatchAction.NETWORK_END_ACTION,
+            ],
+        },
+        {
+            it: 'putObjectTagging: should return expected actions -> test with prefix parameter',
+            fn: actions.putObjectTagging(BUCKET_NAME, PREFIX, OBJECT_KEY, HEAD_OBJECT_RESPONSE, []),
```
Comment (nicolas2bert) [inline src/react/actions/__tests__/s3object.test.jsx]: Can we provide a non-empty tags array? (type `Tags`)

**Review by nicolas2bert — CHANGES_REQUESTED** [review]
First iteration.

File: src/react/actions/__tests__/s3object.test.jsx (hunk 3 above)
Comment (romainlancia) [reply src/react/actions/__tests__/s3object.test.jsx]: done

File: src/react/actions/__tests__/s3object.test.jsx (hunk 2 above)
Comment (romainlancia) [reply src/react/actions/__tests__/s3object.test.jsx]: done

File: src/react/actions/__tests__/s3object.test.jsx (hunk 1 above)
Comment (romainlancia) [reply src/react/actions/__tests__/s3object.test.jsx:95]: done

File: src/js/mock/S3Client.js [hunk 4]
```diff
@@ -79,6 +90,27 @@ export const listBucketsResponse = (prefixWithSlash: string): ListObjectsRespons
     StartAfter: '',
 });
 
+export const userMetadata = {
+    'cache-control': '1',
```
Comment (nicolas2bert) [inline src/js/mock/S3Client.js]: `cache-control` is one of the system metadata. Do you mind using `'color': 'green'` instead?

**Review by nicolas2bert — APPROVED** [review]
One small comment and it is good to get merged.

**Comment by romainlancia** [comment]
@nicolas2bert can you ask for another person to review ? :slightly_smiling_face:

File: src/js/mock/S3Client.js (hunk 4 above)
Comment (romainlancia) [reply src/js/mock/S3Client.js]: done

**Review by terrydervaux — APPROVED** [review]
LGTM

**Comment by romainlancia** [comment]
@bert-e approve

**Pull request merged**